### PR TITLE
Add IHC8223AL

### PR DESCRIPTION
**Pull request by 65698888**
IHC8223AL(z2m用)

### CODE REVIEW

**Comment by 65698888** [comment]
I refer to the writing method in tuya.ts, why can't mine pass.

**Comment by 65698888** [comment]
Was it in JavaScript format before and not modified to TypeScript format?

File: src/devices/honyar.ts [hunk 1]
```diff
@@ -128,6 +181,36 @@ const definitions: Definition[] = [
             await reporting.onOff(endpoint3);
         },
     },
+	{
+		zigbeeModel: ['000a0abb\u0000', 'RH5000_SmartOutlet'],
+		model: 'IHC8223AL',
+		vendor: 'Honyar',
+		description: 'Smart Power Socket 10A (with power monitoring)',
+		fromZigbee: [fz.on_off, fzLocal.honyer_electrical_measurement, fzLocal.honyer_metering],
```
Comment (Koenkk) [inline src/devices/honyar.ts:189]: Could you use modernExtend instead here? Example:

https://github.com/Koenkk/zigbee-herdsman-converters/blob/3553ddedf8cf4ae3751df924b7ea7c0babefacfd/src/devices/adeo.ts#L297

Comment (65698888) [reply src/devices/honyar.ts]: Continue to try to modify, this time there is no error. Please help to check if there are any mistakes.

Comment (Koenkk) [reply src/devices/honyar.ts]: I don't see the `extend` in this PR yet

Comment (65698888) [reply src/devices/honyar.ts]: Can't I do it the way I'm doing it?

Comment (Koenkk) [reply src/devices/honyar.ts]: No, only modernExtend is accepted since this reduces code.

Comment (65698888) [reply src/devices/honyar.ts]: I use modernExtend for the first time, but I am not sure whether it is successful. Please help to check it.

File: src/devices/honyar.ts [hunk 2]
```diff
@@ -128,6 +129,38 @@ const definitions: Definition[] = [
...
+    {
+        zigbeeModel: ['000a0abb\u0000', 'RH5000_SmartOutlet'],
+        model: 'IHC8223AL',
+        vendor: 'Honyar',
+        description: 'Smart Power Socket 10A (with power monitoring)',
+        extend: [
+            onOff({powerOnBehavior: false}),
+            electricityMeter({cluster: 'honyar'}),
+        ],
+        onEvent: async (type, data, device) => {
```
Comment (Koenkk) [inline src/devices/honyar.ts:141]: Can you try without this? I hope this device support reporting so polling is not needed.

Comment (65698888) [reply src/devices/honyar.ts:141]: I've already tried. Proactively report a problem, you have to proactively poll.

File: src/converters/fromZigbee.ts [hunk 3]
```diff
@@ -5140,6 +5140,54 @@ const converters2 = {
...
+    honyer_electrical_measurement: {
+        cluster: 'haElectricalMeasurement',
+        type: ['attributeReport', 'readResponse'],
+        options: [
+            exposes.options.calibration('power', 'percentual'), exposes.options.precision('power'),
+            exposes.options.calibration('current', 'percentual'), exposes.options.precision('current'),
+            exposes.options.calibration('voltage', 'percentual'), exposes.options.precision('voltage'),
+        ],
+        convert: (model, msg, publish, options, meta) => {
+            if (meta.device.dateCode === '20170621') {
```
Comment (Koenkk) [inline src/converters/fromZigbee.ts:5169]: If I understand correctly, there is a bug for this particular firmware which has the wrong multiplier/divisors?

Comment (65698888) [reply src/converters/fromZigbee.ts:5169]: Yes, I tried the following method, but it didn't work.

electricityMeter({current: {divisor: 1000}, voltage: {divisor: 1}, power: {divisor: 1}}),

Comment (Koenkk) [reply src/converters/fromZigbee.ts:5169]: Did you try to do a reconfigure afterwards? This should work (use the yellow refresh icon on the frontend to do so)

Comment (65698888) [reply src/converters/fromZigbee.ts]: Thanks for your guidance, the display is normal after using the yellow refresh button on the front.

File: src/converters/fromZigbee.ts [hunk 4]
```diff
@@ -5140,6 +5140,23 @@ const converters2 = {
             return payload;
         },
     } satisfies Fz.Converter,
+    honyer_metering: {
```
Comment (Koenkk) [inline src/converters/fromZigbee.ts:5143]: Can this one also be removed?

File: src/devices/honyar.ts [hunk 5]
```diff
@@ -128,6 +129,38 @@ const definitions: Definition[] = [
...
+        vendor: 'Honyar',
+        description: 'Smart Power Socket 10A (with power monitoring)',
+        extend: [
+            onOff({powerOnBehavior: false}),
+            electricityMeter({cluster: 'honyar', current: {divisor: 1000}, voltage: {divisor: 1}, power: {divisor: 1}}),
+        ],
+        onEvent: async (type, data, device) => {
+            device.skipDefaultResponse = true;
+            const Endpoint = device.getEndpoint(1);
+            if (Endpoint == null) {
```
Comment (Koenkk) [inline src/devices/honyar.ts:144]: When is this `null`? I wouldn't expect this check here.

File: src/devices/honyar.ts [hunk 6]
```diff
@@ -128,6 +129,38 @@ const definitions: Definition[] = [
...
+            } else if (!globalStore.hasValue(device, 'interval')) {
+                const interval = setInterval(async () => {
+                    try {
+                        await Endpoint.read('haElectricalMeasurement', ['activePower', 'rmsCurrent', 'rmsVoltage']);
+                        await Endpoint.read('seMetering', ['currentSummDelivered']);
+                        await Endpoint.read('genOnOff', ['onOff']);
+                    } catch (error) {
+                        // Do nothing
+                    }
+                }, 5*1000);// Every 5 seconds
```
Comment (Koenkk) [inline src/devices/honyar.ts:159]: Every 5 seconds is too much, default it to 60 seconds and make it configurable (search for `measurement_poll_interval` in this repo for examples)

File: src/devices/honyar.ts (hunk 2 above)
Comment (Koenkk) [reply src/devices/honyar.ts:141]: Can you try reusing existing polling code?

Try with:

```suggestion
        options: [exposes.options.measurement_poll_interval()],
        onEvent: (type, data, device, options) => tuya.onEventMeasurementPoll(type, data, device, options),
```

File: src/converters/fromZigbee.ts (hunk 4 above)
Comment (Koenkk) [reply src/converters/fromZigbee.ts:5143]: @65698888 could you check if this one can be removed?

**Comment by 65698888** [comment]
@Koenkk Hello, is there anything that needs to be modified?

**Comment by Koenkk** [comment]
Yes, please check [this comment](https://github.com/Koenkk/zigbee-herdsman-converters/pull/7494/files#r1616323967)

**Pull request closed without merging**